packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -18,26 +21,27 @@ const sfdxCoreExports = sfdxCoreExtension
   : undefined;
 const { OrgAuthInfo, channelService } = sfdxCoreExports;
 
-// This should be exported from soql-builder-ui
+// TODO: This should be exported from soql-builder-ui
 export interface SoqlEditorEvent {
   type: string;
   payload?: string | string[];
 }
 
-// This should be shared with soql-builder-ui
+// TODO: This should be shared with soql-builder-ui
 export enum MessageType {
   UI_ACTIVATED = 'ui_activated',
   UI_SOQL_CHANGED = 'ui_soql_changed',
   SOBJECT_METADATA_REQUEST = 'sobject_metadata_request',
   SOBJECT_METADATA_RESPONSE = 'sobject_metadata_response',
   SOBJECTS_REQUEST = 'sobjects_request',
   SOBJECTS_RESPONSE = 'sobjects_response',
-  TEXT_SOQL_CHANGED = 'text_soql_changed'
+  TEXT_SOQL_CHANGED = 'text_soql_changed',
+  RUN_SOQL_QUERY = 'run_query'
 }
 
 async function withSFConnection(f: (conn: Connection) => void): Promise<void> {
-  const conn = await OrgAuthInfo.getConnection();
   try {
+    const conn = await OrgAuthInfo.getConnection();
```

Comment:
would the new workspace context changes to preserve the connection across a session be useful here? cc @brpowell

Reply:
Indeed - @jonnyhork check out the new (like yesterday new) `WorkspaceContext` service in core, it takes care of caching the connection for subsequent requests and allows you to listen when the default org changes. You might need to export it from the core api in `index.ts` to use it first.

Reply:
@dobladez You were refactoring the way we use the connection object based on previous feedback right? Does this look right to you?

Reply:
ok, if its that new then, we probably missed it, I'll check it out.

Reply:
@brpowell We are now using the new hotness, see the `fix(soql): use the WorkspaceContext for connection` commit